src/context/AuthContext.tsx

```diff
@@ -0,0 +1,42 @@
+import { createContext, useState, useEffect, ReactNode } from "react";
+import { type User, type LoginCredentials, type SignupCredentials, type AuthContext as AuthContextType } from "../lib/types";
+import { login as loginApi, signup as signupApi, logout as logoutApi, getCurrentUser as getCurrentUserApi } from "../utils/auth";
+
+// Create the context
+export const AuthContext = createContext<AuthContextType | undefined>(undefined);
+
+// Create the Provider Component
+export const AuthProvider = ({ children }: {
+    children: ReactNode
+}) => {
+    const [user, setUser] = useState<User | null>(null);
+
+    const login = async (credentials: LoginCredentials) => {
```

Comment:
keep only the data that needs to be share the state across components in context. Login and signup can be handled locally

Reply:
I removed other functions and only kept the user data in state.